examples/proc_combined_ODNP.py

```diff
@@ -0,0 +1,332 @@
+    read_dict = thislog.log_dict
+for j in range(len(read_array)):
+    thistime,thisrx,thispower,thiscmd = read_array[j]
+fig, (ax_Rx,ax_power) = plt.subplots(2,1, figsize=(10,8))
+fl.next("log figure",fig=fig)
+ax_Rx.set_ylabel('Rx/mV')
+log_start_time = read_array['time'][0]
+print("LOG START TIME IS:",log_start_time)
+relative_time = read_array['time']
+ax_Rx.plot(relative_time, read_array['Rx'],'.')
+ax_power.set_ylabel('power/dBm')
+ax_power.plot(relative_time,read_array['power'],'.')
+mask = read_array['cmd'] != 0
+n_events = len(relative_time[mask])
+trans_power = transforms.blended_transform_factory(
+        ax_power.transData,ax_power.transAxes)
+trans_Rx = transforms.blended_transform_factory(
+        ax_Rx.transData, ax_Rx.transAxes)
+for j, thisevent in enumerate(read_array[mask]):
+    ax_Rx.axvline(x=thisevent['time']-log_start_time)
+    ax_power.axvline(x=thisevent['time']-log_start_time)
+    y_pos = j/n_events
+    #ax_Rx.text(thisevent['time']-log_start_time,y_pos,read_dict[thisevent['cmd']],transform=trans_Rx)
+    #ax_power.text(thisevent['time']-log_start_time,y_pos,read_dict[thisevent['cmd']],transform=trans_power)
+ax_power.legend(**dict(bbox_to_anchor=(1.05,1),loc=2,borderaxespad=0.))
+plt.tight_layout()
+power_axis = nddata(read_array['power'],[-1],['time'])
+power_axis.setaxis('time',relative_time)
+power_axis.setaxis('time',lambda x: x - log_start_time)
+fl.next('power axis')
+fl.plot(power_axis,'.')
+power_axis = nddata(read_array['power'],[-1],['time'])
+power_axis.setaxis('time',relative_time)
+power_axis.name('power')
+fl.next('power axis W')
+coupler_atten = 22
+power_axis.data = (10**((power_axis.data+coupler_atten)/10+3))/1e6 #convert to W
+fl.plot(power_axis,'.')
+#}}}
+#{{{IR processing
+T1_list = []
+power_list = []
+start_times = []
+stop_times = []
+errors=[]
+for nodename, postproc, signal_pathway, clock_correction in [
+        ('FIR_noPower','spincore_IR_v1',{'ph1':0,'ph2':1},False),
+        ('FIR_27dBm','spincore_IR_v1',{'ph1':0,'ph2':1},False),
+        ('FIR_30dBm','spincore_IR_v1',{'ph1':0,'ph2':1},False),
+        ('FIR_32dBm','spincore_IR_v1',{'ph1':0,'ph2':1},False),
+        ('FIR_33dBm','spincore_IR_v1',{'ph1':0,'ph2':1},False),
+        ]:
+    IR = find_file(filename,exp_type=file_location,expno=nodename,
+            postproc=postproc,lookup=lookup_table)
+    times = IR.C.getaxis('indirect').copy()
+    IR.mean('indirect')
+    fl.next('Raw IR')
+    fl.image(IR.C.setaxis('vd','#').set_units('vd','scan #'))
+    IR['ph2',0]['ph1',0]['t2':0]=0 #kill axial noise
+    IR.ift('t2')
+    IR.ift(['ph1','ph2'])
+    t_start = t_range[-1]/4
+    t_start *= 3
+    rx_offset_corr = IR['t2':(t_start,None)]
+    rx_offset_corr = rx_offset_corr.data.mean()
+    IR -= rx_offset_corr
+    IR.ft('t2')
+    IR.ft(['ph1','ph2'])
+    zero_crossing = abs(select_pathway(IR['t2':IR_f_slice],signal_pathway)).C.sum('t2').argmin('vd',raw_index=True).item()
+    IR=IR['t2':IR_f_slice]
```

Comment:
extra junk

Reply:
frequency filtering is extra junk? the zero crossing is required especially for the higher powers where the signal tends to sometimes flip

Reply:
no, not the frequency filtering -- I highlighted too much.

But the zero crossing, and switching back and forth between phase cycling and coherence domain -- yes.

You don't need the zero crossing.